Instruments/include/SkimmerConfig.h

```diff
@@ -60,8 +63,17 @@ struct FileDescriptor {
         first_input_is_ref(false) {}
 
     FileDescriptor(const std::vector<std::string>& _inputs, double _cross_section) :
-        inputs(_inputs), output(inputs.size() ? inputs.front() : ""), cross_section(_cross_section),
-        first_input_is_ref(false) {}
+        output(inputs.size() ? inputs.front() : ""), cross_section(_cross_section),
+        first_input_is_ref(false) {
+            for(const auto& entry: _inputs){
+                std::size_t pos = entry.find(":");
+                std::string partial = entry.substr(0,pos);
+                if (partial == "part") input_is_partial.emplace_back(true);
+                else input_is_partial.emplace_back(false);
+                std::string str = entry.substr(pos+1);
```

Comment:
For the case if pos not found, this works only because of unsigned overflow, better to write differently.